src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertyAccessorSymbol.cs

```diff
@@ -759,7 +759,7 @@ private ImmutableArray<ParameterSymbol> ComputeParameters(BindingDiagnosticBag d
                 if (propertyType.IsStatic)
```

Comment:
>if (propertyType.IsStatic) [](http://example.com/codeflow?start=16&length=26)

It feels like this check belongs in ```SourcePropertySymbolBase.ValidatePropertyType``` method instead. I would explore doing the following:
- report ```WRN_ReturnTypeIsStaticClass``` in ```ValidatePropertyType``` if property has a getter,
- otherwise, report ```WRN_ParameterIsStaticClass``` if property has a setter
- to reduce the noise, suppress ```WRN_ReturnTypeIsStaticClass``` that getter reports for its return type.

As an additional benefit from this approach, we wouldn't need to modify accessibility on ```TypeLocation``` property.

Reply:
Yeah, that sounds much better. Done: a7e81d04fe9470107728afac2f48a6bad9eff8f0